src/Form/AdminSettingsForm.php

```diff
@@ -77,30 +74,43 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#empty' => $this->t('There are no items yet. Add roles.'),
       '#tabledrag' => [
         [
+          'table_id' => 'edit-role-theme-switcher',
           'action' => 'order',
           'relationship' => 'sibling',
           'group' => 'role-order-weight',
         ],
       ],
     ];
 
-    // Generates a form element for each role.
-    $roles = $this->getUserRoles();
-    foreach ($roles as $rid => $role) {
+    $this->setRowPerRole($this->getUserRoles(), $form);
```

Comment:
Generally I prefer use the extract methods & conditional decomposition to avoid all duplication code problems and increase readability. 

In this case we need to isolate this complex and long loop IMHO. We have three major benefits 
- More readable code for builForm method
- Avoid Complexity of buildForm method
- Isolates independent parts of code, meaning that errors are less likely

But I'm not really attach to this point here because we don't have lot complexity (foreach + if / else or wathever...).